docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md

```diff
@@ -158,6 +161,101 @@ The key you provide should look like:
+
+**Generate self-signed certificates**
+Azure AD requires a certificate to sign the SAML requests. You can generate a self-signed certificate using the following command:
+
+```sh
+$ openssl req -x509 -newkey rsa:4096 -keyout key.pem -out cert.pem -days 365 -nodes
+```
+
+This will generate a `key.pem` and `cert.pem` file that you can use for the `private_key` and `certificate` configuration options.
+
+### Add Microsoft Entra SAML Toolkit from the gallery
+
+> Taken from https://learn.microsoft.com/en-us/entra/identity/saas-apps/saml-toolkit-tutorial#add-microsoft-entra-saml-toolkit-from-the-gallery
+
+1. Go to the [Azure portal](https://portal.azure.com/#home) and sign in with your Azure AD account.
+1. Search for `Enterprise Applications`.
+1. In the `Enterprise applications` pane, select `New application`.
+1. In the search box, enter `SAML Toolkit`, and then select the `SAML Toolkit application` from the results panel.
+1. Add a descriptive name and select `Create`.
+
+### Configure the SAML Toolkit application endpoints
+
+In order to validate Azure AD users with Grafana, you need to configure the SAML Toolkit application endpoints by creating a new SAML integration in the Azure AD organization.
+
+> For the following configuration, we will use `https://localhost` as the Grafana URL. Replace it with your Grafana URL.
+
+1. In the `SAML Toolkit application`, select `Set up single sign-on`.
+1. In the `Single sign-on` pane, select `SAML`.
+1. In the Set up `Single Sign-On with SAML` pane, select the pencil icon for Basic SAML Configuration to edit the settings.
+1. In the `Basic SAML Configuration` pane, update the following fields:
+   - In the `Identifier (Entity ID)` field, enter `https://localhost/saml/metadata`.
+   - In the `Identifier (Entity ID)` field, remove the default value.
+   - In the `Reply URL (Assertion Consumer Service URL)` field, enter `https://localhost/saml/acs`.
+   - In the `Sign on URL` field, enter `https://localhost/saml/auth`.
+   - In the `Relay State` field, enter `https://localhost/saml/slo`.
+   - In the `Logout URL` field, enter `https://localhost/logout`.
+1. Select `Save`.
+1. At the `SAML Certificate` section, copy the `App Federation Metadata Url`.
+   - Use this URL in the `idp_metadata_url` field in the `custom.ini` file.
+
+### Configure a Graph API application in Azure AD
+
+While an Azure AD tenant can be configured in Grafana via SAML, some additional information is only accessible via the Graph API. To retrieve this information, you need to create a new application in Azure AD and grant it the necessary permissions.
+
+> [Azure AD SAML limitations](https://learn.microsoft.com/en-us/entra/identity-platform/id-token-claims-reference#groups-overage-claim)
+
+> For the following configuration, we will use `https://localhost` as the Grafana URL. Replace it with your Grafana URL.
+
+#### Create a new Application registration
+
```

Comment:
You should have an intro before the steps. Maybe the text you have on line 229?

Reply:
Intro added as requested